app/src/components/Modals/activity-modal/activity-form-modal.tsx

```diff
@@ -253,30 +242,50 @@ const AddActivityModal: FC<AddActivityModalProps> = ({
       onClose();
       resetSelectedActivityValue();
     } else {
-      toast({
-        status: "error",
-        title: t("activity-value-error"),
-      });
+      const error = response.error as FetchBaseQueryError;
+      const errorData = error.data as any;
+      if (errorData.error?.type === "ManualInputValidationError") {
+        handleManalInputValidationError(errorData.error.issues);
+      } else {
+        toast({
+          status: "error",
+          title: t("activity-value-error"),
+        });
+      }
     }
   };
 
-  let fields = null;
-  let units = null;
-  if (methodology?.id.includes("direct-measure")) {
-    fields = methodology.fields;
-  } else {
-    fields = methodology?.fields[0]["extra-fields"];
-    units = methodology?.fields[0].units;
-  }
+  const closeModalFunc = () => {
+    onClose();
+    resetSelectedActivityValue();
+    reset({
+      activity: {
+        activityType: selectedActivity?.id,
+        fuelType: "",
+        dataQuality: "",
+        sourceReference: "",
+        CH4EmissionFactor: 0,
+        CO2EmissionFactor: 0,
+        N2OEmissionFactor: 0,
+        activityDataAmount: 0,
+        activityDataUnit: null,
+        emissionFactorType: "",
+        totalFuelConsumption: "",
+        totalFuelConsumptionUnits: "",
+        co2EmissionFactorUnit: "",
+        n2oEmissionFactorUnit: "",
+        ch4EmissionFactorUnit: "",
+      },
+    });
```

Comment:
It's helpful to store this as a "default value" object next to the type definition for the form state. Makes it easier if you need to reset the form from different places in the code.

Reply:
Smart idea